tests/DocBlock/TypeExpressionTest.php

```diff
@@ -265,6 +265,101 @@ public static function provideGetConstTypesCases(): iterable
         }
     }
 
+    /**
+     * @dataProvider provideParseInvalidExceptionCases
+     */
+    public function testParseInvalidException(string $value): void
+    {
+        $this->expectException(\Exception::class);
+        $this->expectExceptionMessage('Unable to parse phpdoc type');
+        new TypeExpression($value, null, []);
+    }
+
+    public static function provideParseInvalidExceptionCases(): iterable
+    {
+        yield [''];
+
+        yield ['0_class_cannot_start_with_number'];
```

Comment:
It's self-explaining too when using proper yields' keys, but then it's more friendly in `--testdox` and IDE output. You don't need to duplicate comments, just use proper description as yield key and foo-like data in the actual case. That's the approach widely used in this repo, and in terms of new tests we rather require it.

@kubawerlos WDYT?

Reply:
@Wirone feel free to adjust the names and push them to my PR.

Reply:
> It's self-explaining

Self-explaining code is fine IMHO, especially with small code where in case of failure we will see all of it in CLI.

I have never seen the real profit of using `--testdox`, for me it is just prettier, unnecessary output.

Reply:
For me it's more readable with yield key (some kind of positive lookahead 😉), but I won't insist 🤷‍♂️.